src/blenderbim/blenderbim/bim/module/pset/operator.py

```diff
@@ -374,3 +373,114 @@ def _execute(self, context):
                 prop_name=self.name,
                 prop_value=prop_value,
             )
+
+
+class BIM_OT_add_property_to_edit(bpy.types.Operator):
+    bl_label = "Add new row"
+    bl_idname = "bim.add_property_to_edit"
+    bl_options = {"REGISTER", "UNDO"}
+    option: bpy.props.StringProperty()
+
+    def execute(self, context):
+        getattr(context.scene, self.option).add()
+        return {"FINISHED"}
+
+
+class BIM_OT_remove_property_to_edit(bpy.types.Operator):
+    bl_label = "Remove property to be renamed"
+    bl_idname = "bim.remove_property_to_edit"
+    bl_options = {"REGISTER", "UNDO"}
+    index: bpy.props.IntProperty()
+    option: bpy.props.StringProperty()
+
+    def execute(self, context):
+        getattr(context.scene, self.option).remove()
+        return {"FINISHED"}
+
+
+class BIM_OT_clear_list(bpy.types.Operator):
+    bl_label = "Clear list of properties"
+    bl_idname = "bim.clear_list"
+    bl_options = {"REGISTER", "UNDO"}
+    option: bpy.props.StringProperty()
+
+    def execute(self, context):
+        getattr(context.scene, self.option).clear()
+        return {"FINISHED"}
```

Comment:
I could probably combine the following 3 operator: (BIM_OT_add_property_to_edit, BIM_OT_remove_property_to_edit, BIM_OT_clear_list) into a single operator and just add another attribute "action" that expects either "add, delete or clear".

Reply:
You can do that but in my experience it tends to backfire and makes for more complicated operators when you have to extend them down the line :)